CRM/Report/Form.php

```diff
@@ -4672,6 +4680,10 @@ public function getPermissionedFTQuery(&$query, $alias = NULL, $return = FALSE)
     }
     $financialTypes = NULL;
     CRM_Financial_BAO_FinancialType::getAvailableFinancialTypes($financialTypes);
+    $this->financialTypes = $financialTypes;
```

Comment:
@eileenmcnaughton @colmanw Should we be defining a CRM/Report/ContributionForm.php subclass of CRM/Report/Form.php so only reports needing financial type ACLs get this function and the related $financialTypes variable defined into them? Alternatively, since it will seem odd for a Contact Detail form to be a subclass of a contribution report class, perhaps we should be putting the function into an include file. But before recommending either approach, I want to know which you think is used most, and which is most appropriate.